opnsense/resource_firewall_alias.go

```diff
@@ -0,0 +1,317 @@
+				Optional:    true,
+				Default:     true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of the alias",
+				Required:    true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Description:  "Type of the alias",
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"host", "network", "port", "url"}, false),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of the alias",
+				Optional:    true,
+				Default:     "",
+			},
+			"content": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The content of this alias (IP, Cidr, url, ...)",
+
+				Optional: true,
+			},
+			// TODO add other fields (like proto)
+		},
+	}
+}
+
+func resourceFirewallAliasRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] Getting OPNsense client from meta")
+	c := meta.(*opnsense.Client)
+
+	log.Printf("[TRACE] Converting ID to UUID")
+	uuid, err := uuid.FromString(d.Id())
+	if err != nil {
+		log.Printf("[ERROR]resourceFirewallAliasRead -  Failed to parse ID")
+		return err
+	}
+
+	log.Printf("[TRACE] Fetching alias configuration from OPNsense")
+	alias, err := c.AliasGet(uuid)
+	if err != nil {
+		// temporary fix for the internal error API when we try to get an unreferenced UIID
+		if err.Error() == "Internal Error status code received" {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("ERROR: \n%#v", err)
+		log.Printf("[ERROR] Failed to fetch uuid: %s", uuid)
+		return err
+	}
+
+	log.Printf("[DEBUG] Configuration from OPNsense: \n")
+	log.Printf("[DEBUG] %#v \n", alias)
+
+	d.SetId(alias.UUID.String())
+	d.Set("enabled", alias.Enabled)
+	d.Set("name", alias.Name)
+	d.Set("type", alias.Type)
+	d.Set("description", alias.Description)
+	d.Set("content", alias.Content)
+	parents := []string{}
+
+	// check if this alias is a member of another alias (nested)
+	alias_list, err := c.AliasGetList()
+	if err != nil {
+		log.Printf("[ERROR]: %v", err)
+		return err
+	}
+
+	for _, nested_alias := range alias_list.Rows {
+		if strings.Contains(nested_alias.Content, alias.Name) {
+			parents = append(parents, nested_alias.UUID)
+		}
+	}
+
+	d.Set("parent", parents)
+
+	return nil
+}
+
+func resourceFirewallAliasCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*opnsense.Client)
+	alias := opnsense.AliasFormat{}
+
+	err := prepareFirewallAliasConfiguration(d, &alias)
+	if err != nil {
+		return err
+	}
+
+	// create the alias
+	uuid_created, err := c.AliasAdd(alias)
+	if err != nil {
+		return err
+	}
+
+	// add the alias to his parent if necessary
+	parent := d.Get("parent")
+	if parent != nil {
+		parent_list := parent.(*schema.Set).List()
+		if len(parent_list) > 0 {
+			addNestedAlias(c, parent_list, alias.Name)
+		}
+	}
+
+	// apply configuration change
+	_, err_apply := c.AliasReconfigure()
+	if err_apply != nil {
+		return err_apply
```

Comment:
```suggestion
	_, err = c.AliasReconfigure()
	if err != nil {
		return err
```
I think it would be more idiomatic if this would follow go's error etiquette